SPECS/ocaml-gettext/ocaml-gettext.spec

```diff
@@ -1,36 +1,52 @@
-%global opt %(test -x %{_bindir}/ocamlopt && echo 1 || echo 0)
-%global __ocaml_requires_opts -i Asttypes -i Parsetree
-%global __ocaml_provides_opts -i Pr_gettext
+# OCaml packages not built on i686 since OCaml 5 / Fedora 39.
+ExcludeArch: %{ix86}
+
+# Optionally disable camomile dep on RHEL.
+%if !0%{?rhel}
+%bcond_without camomile
+%bcond_without tests
+%else
+%bcond_with    camomile
+%bcond_with    tests
+%endif
 
 Summary:        OCaml library for i18n
 Name:           ocaml-gettext
 Version:        0.4.2
-Release:        2%{?dist}
+Release:        3%{?dist}
 License:        LGPLv2+ WITH exceptions
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 URL:            https://github.com/gildor478/ocaml-gettext
-Source0:        https://github.com/gildor478/ocaml-gettext/releases/download/v%{version}/gettext-v%{version}.tbz
+Source0:        %{url}/releases/download/v%{version}/gettext-v%{version}.tbz
+
+# Updates for OCaml 5.  Based in part on
+# https://github.com/gildor478/ocaml-gettext/pull/24
+Patch0:         %{name}-ocaml5.patch
+# Adapt to changes in camomile 2.0
+# https://github.com/gildor478/ocaml-gettext/pull/27
+Patch1:         %{name}-camomile2.patch
 
-BuildRequires:  autoconf
-BuildRequires:  automake
-BuildRequires:  chrpath
+BuildRequires:  ocaml >= 5.1.1
+BuildRequires:  ocaml-fileutils-devel >= 0.4.4-4
+BuildRequires:  ocaml-dune >= 1.11.0
+BuildRequires:  ocaml-dune-configurator-devel
+BuildRequires:  ocaml-cppo
 BuildRequires:  docbook-style-xsl
-BuildRequires:  libxml2
 BuildRequires:  libxslt
-BuildRequires:  ocaml >= 4.00.1
-BuildRequires:  ocaml-camomile-data
-BuildRequires:  ocaml-camomile-devel >= 0.8.6-3
-BuildRequires:  ocaml-compiler-libs
-BuildRequires:  ocaml-cppo
-BuildRequires:  ocaml-dune-devel
-BuildRequires:  ocaml-fileutils-devel >= 0.4.4-4
-BuildRequires:  ocaml-findlib-devel >= 1.3.3-3
-BuildRequires:  ocaml-ocamldoc
+BuildRequires:  libxml2
+%if %{with tests}
```

Comment:
Please use `%{with_check}` for test-only dependencies.

Reply:
Fixed